File: src/my_project/medium_problems/from101to150/remove_duplicates_sorted_array.py
from typing import List 

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        if not nums:
            return 0
        i = 0
        count = 1
        flag = False 
        for j in range(1, len(nums)):
            if nums[j] == nums[i]:
                if not flag: 
                    flag = True
                    i += 1
                    nums[i], nums[j] = nums[j], nums[i]
                    count += 1
            else:
                flag = False
                i += 1
                nums[i], nums[j] = nums[j], nums[i]
                count += 1
        return count
        
        

File: my_project_typescript/easy_problems_typescript/from1to50/removeDuplicatesSortedArray.ts
function removeDuplicates(nums: number[]): number {
    let j: number = 0;
    let numsLenght: number = nums.length;
    if (numsLenght === 0 || numsLenght ===1) {
        return numsLenght;
    }
    for (let i = 0; i < numsLenght -1 ; i++ ){
        if (nums[i] !== nums[i+1]){
            nums[j] = nums[i]
            j += 1
        }
    
    }
    nums[j] = nums[numsLenght-1]
    return j+1;
};


console.log(removeDuplicates([0,0,1,1,1,2,2,3,3,4]));
console.log(removeDuplicates([1,1,2]));
console.log(removeDuplicates([1,2,3]));